pds4indextools/pds4_create_xml_index.py

```diff
@@ -75,42 +114,46 @@ def default_value_for_nil(config, data_type, nil_value):
     return default
 
 
-def convert_header_to_tag(path, root, namespaces):
-    """Convert an XPath expression to an XML tag.
+def filter_dict_by_glob_patterns(input_dict, glob_patterns):
+    """Filter a dictionary based on a list of glob patterns matching for keys.
 
     Inputs:
-        path          XPath expression.
-        root          The root element of the XML document.
-        namespaces    Dictionary of XML namespace mappings.
-
+        input_dict:       The dictionary to filter.
+        glob_patterns:    A list of glob patterns to match against dictionary keys.
+        
     Returns:
-        Converted XML tag.
+        Filtered dictionary with desired contents
     """
-    tag = str(root.xpath(path, namespaces=namespaces)[0].tag)
-
-    return tag
-
-
-def convert_header_to_xpath(root, xpath_find, namespaces):
-    """Convert an XML header path to an XPath expression.
-
-    Inputs:
-        root           The root element of the XML document.
-        xpath_find     Original XML header path.
-        namespaces     Dictionary of XML namespace mappings.
-
-    Returns:
-        Converted XPath expression.
-    """
-    sections = xpath_find.split('/')
-    xpath_final = ''
-    portion = ''
-    for sec in sections[1:]:
-        portion = portion + '/' + sec
-        tag = str(root.xpath(portion, namespaces=namespaces)[0].tag)
-        xpath_final = xpath_final + '/' + tag
+    filtered_dict = {}
+
+    if glob_patterns is not None:
+        if not all(pat.startswith('!') for pat in glob_patterns):
```

Comment:
This doesn't work. The idea is that the glob patterns are executed *in order*. For example:
```
a*
!ab*
abc*
```
would match anything starting with a, unless it started with ab, but starting with abc is still OK. If you break the patterns into separate positive/negative lists and then do all the positives and all the negatives, you don't get this behavior. Just do the patterns in order. The code can be much simpler.

Also, I think you're treating the case of:
```
!a
```
as meaning:
```
*
!a
```
(hence the use of `all()` and the separate if/else) but that's not what the user said. If the user just says:
```
!a
```
then they should get no elements.

Reply:
This has been fixed.